tests/components/compensation/test_sensor.py

```diff
@@ -1,115 +1,209 @@
+    STATE_UNAVAILABLE,
     STATE_UNKNOWN,
+    UnitOfTemperature,
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.setup import async_setup_component
 
+from tests.common import assert_setup_component, get_fixture_path
+
+TEST_OBJECT_ID = "test_compensation"
+TEST_ENTITY_ID = "sensor.test_compensation"
+TEST_SOURCE = "sensor.uncompensated"
+
+TEST_BASE_CONFIG = {
+    "source": TEST_SOURCE,
+    "data_points": [
+        [1.0, 2.0],
+        [2.0, 3.0],
+    ],
+    "precision": 2,
+}
+TEST_CONFIG = {
+    "name": TEST_OBJECT_ID,
+    "unit_of_measurement": "a",
+    **TEST_BASE_CONFIG,
+}
 
-async def test_linear_state(hass: HomeAssistant) -> None:
-    """Test compensation sensor state."""
-    config = {
-        "compensation": {
-            "test": {
-                "source": "sensor.uncompensated",
-                "data_points": [
-                    [1.0, 2.0],
-                    [2.0, 3.0],
-                ],
-                "precision": 2,
-                "unit_of_measurement": "a",
-            }
-        }
-    }
-    expected_entity_id = "sensor.compensation_sensor_uncompensated"
 
-    assert await async_setup_component(hass, DOMAIN, config)
-    assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+async def async_setup_compensation(
+    hass: HomeAssistant, count: int, config: dict[str, Any]
+) -> None:
+    """Do setup of a compensation integration sensor."""
+    with assert_setup_component(count, DOMAIN):
+        assert await async_setup_component(
+            hass,
+            DOMAIN,
+            {DOMAIN: {"test": config}},
+        )
+
     await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
 
+@pytest.fixture
+async def setup_compensation(
+    hass: HomeAssistant, count: int, config: dict[str, Any]
+) -> None:
+    """Do setup of a compensation integration sensor."""
+    await async_setup_compensation(hass, count, config)
+
+
+@pytest.fixture
+async def setup_compensation_with_extra_config(
+    hass: HomeAssistant,
+    count: int,
+    config: dict[str, Any],
+    extra_config: dict[str, Any],
+) -> None:
+    """Do setup of a compensation integration sensor."""
+    await async_setup_compensation(hass, count, {**config, **extra_config})
+
+
+@pytest.fixture
+async def setup_compensation_with_limits(
+    hass: HomeAssistant,
+    count: int,
+    config: dict[str, Any],
+    upper: bool,
+    lower: bool,
+):
+    """Do setup of a compensation integration sensor with extra config."""
+    await async_setup_compensation(
+        hass,
+        count,
+        {
+            **config,
+            "lower_limit": lower,
+            "upper_limit": upper,
+        },
+    )
+
+
+@pytest.fixture
+async def caplog_setup_text(caplog: pytest.LogCaptureFixture) -> str:
+    """Return setup log of integration."""
+    return caplog.text
+
+
+@pytest.mark.parametrize(
+    ("count", "config"),
+    [(1, TEST_CONFIG)],
+)
```

Comment:
```suggestion
@pytest.mark.parametrize("config", [TEST_CONFIG])
```

Please adjust all other tests